code/test/util.js

```diff
@@ -215,6 +269,123 @@ const getEOTP = async ({ seed, hseed, effectiveTime, timeOffset }) => {
   return { index, eotp }
 }
 
+// transactionExecute commits and reveals a wallet transaction
+const transactionExecute = async ({ wallet, operationType, tokenType, contractAddress, tokenId, dest, amount, data, address, randomSeed, backlinkAddresses, testTime }) => {
+  if (testTime === undefined) { testTime = Date.now() }
+  // // calculate counter from testTime
+  const counter = Math.floor(testTime / INTERVAL)
+  const otp = ONEUtil.genOTP({ seed: wallet.seed, counter })
+  // // calculate wallets effectiveTime (creation time) from t0
+  const info = await wallet.wallet.getInfo()
+  const t0 = new BN(info[3]).toNumber()
+  const walletEffectiveTime = t0 * INTERVAL
+  const index = ONEUtil.timeToIndex({ effectiveTime: walletEffectiveTime, time: testTime })
+  const eotp = await ONE.computeEOTP({ otp, hseed: wallet.hseed })
+
+  // Format commit and revealParams based on tokenType
+  let commitParams
+  let revealParams
+  let paramsHash
+  console.log(`operationType: ${operationType}`)
+  switch (operationType) {
+    case ONEConstants.OperationType.TRACK:
+    case ONEConstants.OperationType.UNTRACK:
+      paramsHash = ONEWallet.computeGeneralOperationHash
+      commitParams = { operationType, tokenType, contractAddress, dest }
+      revealParams = { operationType, tokenType, contractAddress, dest }
+      break
+    case ONEConstants.OperationType.OVERRIDE_TRACK:
+      paramsHash = ONEWallet.computeGeneralOperationHash
+      commitParams = { operationType, tokenType, contractAddress, tokenId, dest, amount, data }
+      revealParams = { operationType, tokenType, contractAddress, tokenId, dest, amount, data }
+      break
+    case ONEConstants.OperationType.RECOVER_SELECTED_TOKENS:
+    case ONEConstants.OperationType.BATCH:
+      paramsHash = ONEWallet.computeDataHash
+      commitParams = { operationType, data }
+      revealParams = { operationType, data }
+      break
+    case ONEConstants.OperationType.BACKLINK_ADD:
+    case ONEConstants.OperationType.BACKLINK_DELETE:
+    case ONEConstants.OperationType.BACKLINK_OVERRIDE:
+      paramsHash = ONEWallet.computeDataHash
+      commitParams = { operationType, backlinkAddresses, data }
+      revealParams = { operationType, backlinkAddresses, data }
+      break
+    case ONEConstants.OperationType.COMMAND:
+      paramsHash = ONEWallet.computeDataHash
```

Comment:
most commit and reveal params below do not match the hash functions. Some operation types do not match corresponding commit hash functions (should not be called "paramsHash"). This whole function needs some major refactoring